tests/testthat/test-object_usage_linter.R

```diff
@@ -624,3 +624,17 @@ test_that("messages without a quoted name are caught", {
     object_usage_linter()
   )
 })
+
+test_that("functional lambda definitions are also caught", {
+  skip_if_not_r_version("4.1.0")
+
+  expect_lint(
+    trim_some("
+      fun <- \\() {
+        a <- 1
+      }
+    "),
+    rex::rex("local variable", anything, "assigned but may not be used"),
```

Comment:
Why is it `anything` and not `'a'` here?

Reply:
Just copied it from the above test:

https://github.com/r-lib/lintr/blob/4f12a43d525994bf783e037605941d7bbbfd2c03/tests/testthat/test-object_usage_linter.R#L3

The advantage I see is we don't need to bother about whether `a` is quoted, or with which characters. OTOH that above usage may just be to re-use the message across tests where different-named variables are linted.

Happy to change if you'd like.

Reply:
I think it's fine as it is then.